### PR TITLE
[mempool] Mempool testing

### PR DESCRIPTION
**Pull request by gregnazario**
Testing using cluster test

### CODE REVIEW

File: mempool/src/counters.rs [hunk 1]
```diff
@@ -214,19 +246,32 @@ pub static SHARED_MEMPOOL_PENDING_BROADCASTS_COUNT: Lazy<IntGaugeVec> = Lazy::ne
...
 
-pub static SHARED_MEMPOOL_TRANSACTIONS_PROCESSED: Lazy<IntCounterVec> = Lazy::new(|| {
+pub fn shared_mempool_pending_broadcasts(peer: &PeerNetworkId) -> IntGauge {
+    SHARED_MEMPOOL_PENDING_BROADCASTS_COUNT.with_label_values(&[
+        peer.raw_network_id().as_str(),
+        peer.peer_id().short_str().as_str(),
+    ])
+}
+
+static SHARED_MEMPOOL_TRANSACTIONS_PROCESSED: Lazy<IntCounterVec> = Lazy::new(|| {
```
Comment (davidiw) [inline mempool/src/counters.rs:256]: inconsistent ordering of functions and counters

File: mempool/src/shared_mempool/coordinator.rs [hunk 2]
```diff
@@ -75,128 +74,22 @@ pub(crate) async fn coordinator<V>(
...
-                        sample!(
-                            SampleRate::Duration(Duration::from_secs(60)),
-                            warn!(
-                                LogSchema::new(LogEntry::UnexpectedNetworkMsg)
-                                    .peer(&PeerNetworkId(network_id, peer_id))
-                            )
-                        );
-                    }
-                }
+                handle_event(&executor, &bounded_executor, &mut scheduled_broadcasts, &mut smp, network_id, event).await;
```
Comment (davidiw) [inline mempool/src/shared_mempool/coordinator.rs:92]: handle network event? also you probably noticed this, but this isn't actually the network interface, it is some extra thing on top of it. (I think)

File: mempool/src/shared_mempool/coordinator.rs [hunk 3]
```diff
@@ -75,128 +74,22 @@ pub(crate) async fn coordinator<V>(
...
-                    .start_timer();
-                bounded_executor
-                .spawn(tasks::process_client_transaction_submission(
-                    smp.clone(),
-                    msg,
-                    callback,
-                    task_start_timer,
-                ))
-                .await;
+                handle_client_event(&mut smp, &bounded_executor, msg, callback).await;
```
Comment (davidiw) [inline mempool/src/shared_mempool/coordinator.rs:77]: handle_client_request?

**Review by davidiw — COMMENTED** [review]
this is blocked by the lack of a test for the default fn failover. looks good otherwise

**Comment by gregnazario** [comment]
> this is blocked by the lack of a test for the default fn failover. looks good otherwise

@davidiw This is kinda a clustering of 2-3 different PRs, because I'm just holding one PR open so I can run cluster test against whatever I want.  Hence the `draft` :)

**Pull request closed without merging**